diff -Nur netkit-telnet-0.17.orig/configure netkit-telnet-0.17.netkit-telnet-fixes/configure
--- netkit-telnet-0.17.orig/configure	2000-07-29 20:00:29.000000000 +0200
+++ netkit-telnet-0.17.netkit-telnet-fixes/configure	2006-08-05 15:33:01.000000000 +0200
@@ -57,7 +57,8 @@
 
 BINDIR="$EXECPREFIX/bin"
 SBINDIR="$EXECPREFIX/sbin"
-MANDIR="$PREFIX/man"
+DATADIR="$PREFIX/share"
+MANDIR="$DATADIR/man"
 
 echo "Directories: $BINDIR $SBINDIR $MANDIR "
 
@@ -168,7 +169,6 @@
          $CC -O2 __conftest.c -o __conftest
        ) >/dev/null 2>&1; then
          echo 'yes'
-         CFLAGS="$CFLAGS -O2"
     else
          echo 'no'
          echo -n "Checking if $CC accepts -O... "
@@ -202,7 +202,6 @@
              $CXX -O2 __conftest.cc -o __conftest
            ) >/dev/null 2>&1; then
              echo 'yes'
-             CXXFLAGS="$CXXFLAGS -O2"
         else
              echo 'no'
              echo -n "Checking if $CXX accepts -O... "
@@ -252,7 +251,6 @@
 fi
 
 
-LDFLAGS=
 LIBS=
 
 rm -f __conftest*
@@ -544,6 +542,7 @@
 
     echo "BINDIR=$BINDIR"
     echo "SBINDIR=$SBINDIR"
+    echo "DATADIR=$DATADIR"
     echo "MANDIR=$MANDIR"
     echo "BINMODE=$BINMODE"
     echo "DAEMONMODE=$DAEMONMODE"
